firefox.launcher

```diff
@@ -19,37 +19,50 @@
 # This requires the personal-files plug to be connected.
 # This is a stopgap measure until proper profile migration is implemented
 # in firefox (see https://bugzilla.mozilla.org/1730530).
-DOTMOZILLA="$SNAP_USER_COMMON/.mozilla"
+
+export XDG_CONFIG_HOME="$SNAP_USER_COMMON/.config"
```

Comment:
Likely we want:
 - setting `XDG_CONFIG_HOME` in the `snapcraft.yaml` directly
 - thus allow override at https://github.com/snapcore/snapcraft-desktop-integration/blob/a3df144a8bf90b9e35ec1b05a9feea2f3069fa81/common/init#L42